tests/src/test/java/alluxio/server/ft/journal/TableMasterJournalIntegrationTest.java

```diff
@@ -0,0 +1,132 @@
+import alluxio.table.common.udb.UnderDatabaseRegistry;
+import alluxio.table.under.hive.HiveDatabaseFactory;
+import alluxio.table.under.hive.HivePartition;
+import alluxio.testutils.LocalAlluxioClusterResource;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Integration tests for table master functionality.
+ */
+@RunWith(PowerMockRunner.class)
+@PowerMockIgnore({"javax.crypto.*", "javax.security.*", "sun.security.*", "javax.net.ssl.*"})
+@PrepareForTest(AlluxioCatalog.class)
+public class TableMasterJournalIntegrationTest {
+  @Rule
+  public LocalAlluxioClusterResource mClusterResource =
+      new LocalAlluxioClusterResource.Builder().setStartCluster(false)
+          .build();
+
+  private static final String DB_NAME = "testdb";
+  private static final String TABLE_NAME_PREFIX = "testtable";
+
+  private void createMockUdbAndTable(int numOfTables) throws Exception {
+    UnderDatabase udb = PowerMockito.mock(UnderDatabase.class);
+    when(udb.getName()).thenReturn(DB_NAME);
+    String[] tableNames = new String[numOfTables];
+    Arrays.setAll(tableNames, i -> TABLE_NAME_PREFIX + Integer.toString(i));
+    when(udb.getTableNames()).thenReturn(Arrays.asList(tableNames));
+
+    for (String tableName : tableNames) {
+      UdbTable udbTable = PowerMockito.mock(UdbTable.class);
+      when(udbTable.getName()).thenReturn(tableName);
+      when(udbTable.getOwner()).thenReturn("test");
+      when(udbTable.getPartitionCols()).thenReturn(Collections.emptyList());
+
+      when(udbTable.getLayout()).thenReturn(Layout.getDefaultInstance());
+      when(udbTable.getParameters()).thenReturn(Collections.emptyMap());
+      when(udbTable.getSchema()).thenReturn(Schema.getDefaultInstance());
+      when(udbTable.getPartitions()).thenReturn(Arrays.asList(new HivePartition(
+          new HiveLayout(PartitionInfo.newBuilder().setDbName(DB_NAME)
+              .setTableName(tableName).setStorage(Storage.getDefaultInstance()).build(),
+              Collections.emptyList()))));
+      when(udb.getTable(tableName)).thenReturn(udbTable);
+    }
+
+    UnderDatabaseRegistry registry = PowerMockito.mock(UnderDatabaseRegistry.class);
+
+    doNothing().when(registry).refresh();
+    when(registry.create(any(UdbContext.class), anyString(), any(UdbConfiguration.class)))
+        .thenReturn(udb);
+    PowerMockito.whenNew(UnderDatabaseRegistry.class).withNoArguments().thenReturn(registry);
+  }
+
+  @Before
+  public void before() throws Exception {
+    createMockUdbAndTable(1);
+  }
+
+  @Test
+  public void journalAttachDb() throws Exception {
```

Comment:
Shouldn't we test additional operations like detach, new layouts, etc?

Reply:
yeah that will come once we settle on the testing infrastructure.